src/main/java/com/google/sps/servlets/NewsServlet.java

```diff
@@ -0,0 +1,114 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.Article;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Arrays;
+import java.util.ArrayList;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.annotation.WebServlet;
+import com.google.sps.data.UrlRequest;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/** Servlet that gets news information. */
+@WebServlet("/news")
+public class NewsServlet extends HttpServlet {
+
+  public final String GOOGLE_NEWS_PATH = "https://news.google.com/search";
+  public final String articleTag = "DY5T1d";  // Class tag for articles in Google News HTML code.
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> terms = Arrays.asList(request.getParameterValues("key"));
+    HashMap<String, List<Article>> articleMap = new HashMap<>();
+    terms.forEach((term) -> {
+      try {
+        Map<String, String> paramMap = new HashMap<>();
+        paramMap.put("q", UrlRequest.encodeTerm(term));
+        paramMap.put("hl", "en-US");
```

Comment:
[Optional] Might want to allow user to specify these params in the future.

Reply:
Or possibly set defaults based on user's location.

Reply:
I am going to make a new branch for this.